hack/release/go.mod

```diff
@@ -1,26 +1,41 @@
 module github.com/Mellanox/network-operator-docs
 
-go 1.21
+go 1.23.0
+
+toolchain go1.23.2
```

Comment:
nit: do you need specifically 1.23.2 or newer ?  i think you can drop this directive.

Reply:
Removed